<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>614</width>
    <height>453</height>
   </rect>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>My Linear Algebra</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>matrix_icon_137416.png</normaloff>matrix_icon_137416.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QLabel" name="lable_title">
        <property name="font">
         <font>
          <family>Script MT</family>
          <pointsize>20</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="focusPolicy">
         <enum>Qt::FocusPolicy::NoFocus</enum>
        </property>
        <property name="text">
         <string>My Linear Algebra</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="lable_matrix">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>18</pointsize>
                <italic>true</italic>
                <bold>false</bold>
               </font>
              </property>
              <property name="text">
               <string>Input your matrix here...</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_rows">
                <property name="text">
                 <string>Rows</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEdit_rows"/>
              </item>
              <item>
               <widget class="QLabel" name="label_columns">
                <property name="text">
                 <string>Columns</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEdit_columns"/>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTableWidget" name="InputMatrix"/>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="lable_matrix_2">
              <property name="font">
               <font>
                <family>Times New Roman</family>
                <pointsize>18</pointsize>
                <italic>true</italic>
               </font>
              </property>
              <property name="text">
               <string>Get your results here...</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextBrowser" name="OutputMatrix"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QComboBox" name="qcb_operation">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="font">
             <font>
              <family>Times New Roman</family>
              <pointsize>14</pointsize>
              <italic>true</italic>
             </font>
            </property>
            <property name="currentText">
             <string>Choose one operation</string>
            </property>
            <property name="insertPolicy">
             <enum>QComboBox::InsertPolicy::InsertAlphabetically</enum>
            </property>
            <item>
             <property name="text">
              <string>Choose one operation</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>加载矩阵A(Load Matrix A)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>加载矩阵B(Load Matrix B)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>矩阵乘法 (A * B)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>矩阵转置 (A^T)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>高斯消元 (Gauss Elimination)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>高斯若尔当消元 (Gauss-Jordan Elimination)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>逆矩阵 (Inversed Matrix)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>行列式计算-莱布尼茨方法(Determination Calculation - Leibniz)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>行列式计算-高斯消元方法(Determination Calculation - Gauss)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>LU 分解 (LU Decomposition - PA = LU)</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>78</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pb_process">
            <property name="font">
             <font>
              <family>Times New Roman</family>
              <pointsize>14</pointsize>
              <italic>true</italic>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Process</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="lable_status">
          <property name="font">
           <font>
            <family>Times New Roman</family>
            <pointsize>14</pointsize>
            <italic>true</italic>
           </font>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LayoutDirection::LeftToRight</enum>
          </property>
          <property name="text">
           <string>Ready...
Matrix A: Unloaded Matrix B: Unloaded</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>614</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMyLinearAlgebra">
    <property name="title">
     <string>MyLinearAlgebra</string>
    </property>
    <widget class="QMenu" name="menuOpen_JSON_Matrix">
     <property name="title">
      <string>Open JSON Matrix...</string>
     </property>
     <addaction name="actionLoad_Matrix_A"/>
     <addaction name="actionLoad_Matrix_B"/>
    </widget>
    <addaction name="menuOpen_JSON_Matrix"/>
   </widget>
   <addaction name="menuMyLinearAlgebra"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_Matrix_A">
   <property name="text">
    <string>Load Matrix A</string>
   </property>
  </action>
  <action name="actionLoad_Matrix_B">
   <property name="text">
    <string>Load Matrix B</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
